src/Maui/Prism.Maui/Navigation/PageNavigationService.cs

```diff
@@ -340,9 +340,7 @@ protected virtual async Task ProcessNavigation(Page currentPage, Queue<string> s
 
         var pageParameters = UriParsingHelper.GetSegmentParameters(nextSegment);
 
-        useModalNavigation =  pageParameters.ContainsKey(KnownNavigationParameters.UseModalNavigation) ? pageParameters.GetValue<bool>(KnownNavigationParameters.UseModalNavigation) : false;        
-        if (!useModalNavigation.Value && !MvvmHelpers.HasNavigationPageParent(currentPage))
```

Comment:
please restore this. However it needs to be updated to:
```cs
if (!useModalNavigation.Value && !MvvmHelpers.HasNavigationPageParent(currentPage) && (currentPage is not FlyoutPage || (currentPage.Parent is FlyoutPage parentFlyout && parentFlyout.Flyout == currentPage)))
```

Reply:
Sure, I will do it.
But a lot of methods that get called afterward have a _nullable_ bool parameter `useModalNavigation`.

If we revert the code to set useModalNavigation _always_ either to true or false, it will never be null. Should we change the `useModalNavigation` parameter of all methods to not nullable? Or do you plan to use it later with nullable?